src/test/ui/impl-trait/must_outlive_least_region_or_bound.stderr

```diff
@@ -1,47 +1,95 @@
-   |              data with this lifetime...
+   |              |        ...is required to live as long as `'static` by this...
+   |              this data with an anonymous lifetime `'_`...
    |
-help: to permit non-static references in an `impl Trait` value, you can add an explicit bound for the anonymous lifetime #1 defined on the function body at 3:1
+help: to permit non-static references in an `impl Trait` value, you can add an explicit bound for an anonymous lifetime `'_`
    |
 LL | fn elided(x: &i32) -> impl Copy + '_ { x }
    |                                 ^^^^
 
-error: cannot infer an appropriate lifetime
+error[E0758]: cannot infer an appropriate lifetime
   --> $DIR/must_outlive_least_region_or_bound.rs:6:44
    |
 LL | fn explicit<'a>(x: &'a i32) -> impl Copy { x }
    |                    -------     ---------   ^ ...and is captured here
    |                    |           |
-   |                    |           ...is required to be `'static` by this...
-   |                    data with this lifetime...
+   |                    |           ...is required to live as long as `'static` by this...
+   |                    this data with lifetime `'a`...
    |
-help: to permit non-static references in an `impl Trait` value, you can add an explicit bound for the lifetime `'a` as defined on the function body at 6:13
+help: to permit non-static references in an `impl Trait` value, you can add an explicit bound for lifetime `'a`
    |
 LL | fn explicit<'a>(x: &'a i32) -> impl Copy + 'a { x }
    |                                          ^^^^
 
-error: cannot infer an appropriate lifetime
-  --> $DIR/must_outlive_least_region_or_bound.rs:12:69
+error[E0758]: cannot infer an appropriate lifetime
+  --> $DIR/must_outlive_least_region_or_bound.rs:9:46
+   |
+LL | fn elided2(x: &i32) -> impl Copy + 'static { x }
+   |               ----     -------------------   ^ ...and is captured here
+   |               |        |
+   |               |        ...is required to live as long as `'static` by this...
+   |               this data with an anonymous lifetime `'_`...
+   |
+help: consider changing the `impl Trait`'s explicit `'static` bound to an anonymous lifetime `'_`
+   |
+LL | fn elided2(x: &i32) -> impl Copy + '_ { x }
+   |                                    ^^
+help: alternatively, set an explicit `'static` lifetime to this parameter
+   |
+LL | fn elided2(x: &'static i32) -> impl Copy + 'static { x }
+   |               ^^^^^^^^^^^^
+
+error[E0758]: cannot infer an appropriate lifetime
+  --> $DIR/must_outlive_least_region_or_bound.rs:12:55
+   |
+LL | fn explicit2<'a>(x: &'a i32) -> impl Copy + 'static { x }
+   |                     -------     -------------------   ^ ...and is captured here
+   |                     |           |
+   |                     |           ...is required to live as long as `'static` by this...
+   |                     this data with lifetime `'a`...
```

Comment:
Again, the vertical ordering here is a little confusing.

Reply:
The third label would normally appear on its own line, always after the other two.

For the first two, I agree it is less than ideal, but because the main span is on `x`, I can't really split this using notes, although I guess we could do the following:

```
error[E0758]: cannot infer an appropriate lifetime
  --> $DIR/must_outlive_least_region_or_bound.rs:3:35
   |
LL | fn elided(x: &i32) -> impl Copy { x }
   |                                   ^ the data captured here...
   |
note: has an anonymous lifetime `'_`
   |
LL | fn elided(x: &i32) -> impl Copy { x }
   |              ^^^^
note: ...but is required to live as long as `'static` by this
   |
LL | fn elided(x: &i32) -> impl Copy { x }
   |                       ^^^^^^^^^
help: to permit non-static references in an `impl Trait` value, you can add an explicit bound for an anonymous lifetime `'_`
   |
LL | fn elided(x: &i32) -> impl Copy + '_ { x }
   |                                 ^^^^
```

Reply:
This is what this currently looks like (with realistic formatting):

<img width="824" alt="Screen Shot 2020-06-01 at 4 18 50 PM" src="https://user-images.githubusercontent.com/1606434/83463680-a064e980-a423-11ea-967b-df5e35ecc0bf.png">

I could tweak it to be

```
error[E0758]: cannot infer an appropriate lifetime
 --> src/test/ui/impl-trait/must_outlive_least_region_or_bound.rs:4:5
  |
3 | fn elided(x: &i32) -> impl Copy {
  |              ----
  |              |
  |              this data with an anonymous lifetime `'_`...
4 |     x
  |     ^ ...is captured here...
  |
note: ...and is required to live as long as `'static` here
  |
3 | fn elided(x: &i32) -> impl Copy {
  |                       ^^^^^^^^^
help: to declare that the `impl Trait` captures data from argument `x`, you can add an explicit `'_` lifetime bound
  |
3 | fn elided(x: &i32) -> impl Copy + '_ {
  |                                 ^^^^
```

Reply:
The current behaviour is better, I think.

Reply:
The last commit has this output:

<img width="826" alt="Screen Shot 2020-06-01 at 5 51 54 PM" src="https://user-images.githubusercontent.com/1606434/83468091-a31a0b80-a430-11ea-9f27-09ed7941fc54.png">